src/aoc2020/day2_copy.clj

```diff
@@ -0,0 +1,152 @@
+(ns aoc2020.day2-copy
+  (:require [aoc2020.util :refer [file->seq]]
+            [clojure.string :as str]))
+
+(defn parse-policy-password-str
+  [policy-password-str]
+  (let [[policy letter pwd] (str/split policy-password-str #" ")
+        [lo hi] (->> (str/split policy #"-")
+                     (mapv parse-long))
+        letter (first letter)]
+    {:policy {:lo     lo
+              :hi     hi
+              :letter letter}
+     :pwd    pwd}))
+
+(defn valid-password?
+  [policy-password]
+  (let [{:keys [policy pwd]} policy-password
+        {:keys [lo hi letter]} policy
+        freqs (frequencies (seq pwd))
+        {letter-freq letter :or {letter-freq 0}} freqs]
+    (<= lo letter-freq hi)))
+
+(defn count-valid-password
+  [policy-password-str-s]
+  (let [policy-password-s (->> policy-password-str-s
+                               (map parse-policy-password-str))]
+    (->> policy-password-s
+         (filter valid-password?)
+         (count))))
+
+
+
+(defn parse-policy-password-str
+  [policy-password-str]
+  (let [[policy letter pwd] (str/split policy-password-str #" ")
+        [pos1 pos2] (->> (str/split policy #"-")
+                         (mapv parse-long))
+        letter (first letter)]
+    {:policy {:pos1   pos1
+              :pos2   pos2
+              :letter letter}
+     :pwd    pwd}))
+
+(defn valid-password?
+  [policy-password]
+  (let [{:keys [policy pwd]} policy-password
+        {:keys [pos1 pos2 letter]} policy
+        letter-pos1 (get pwd (- pos1 1))
+        letter-pos2 (get pwd (- pos2 1))
+        pos1?       (= letter letter-pos1)
+        pos2?       (= letter letter-pos2)
```

Comment:
The 4 lines above should probably be rewritten with mapv + destructuring. The names pos1 and pos2 are not representative of their meanings